include/SugarEmailAddress/SugarEmailAddress.php

```diff
@@ -1144,21 +1193,50 @@ public function AddUpdateEmailAddress($addr, $invalid = 0, $opt_out = 0, $id = n
             }
         }
 
+        // confirmed opt in check
+        $isValidEmailAddress = ($opt_out !== 1 && $invalid !== 1);
+        $this->retrieve($id);
+        $optInIndication = $this->getOptInIndication();
+        if (
+           $isValidEmailAddress
+           && EmailAddressIndicator::isOptedInStatus($optInIndication)
+           && (int)$optInFlag === 1
+        ) {
+           $new_confirmed_opt_in = $this->confirm_opt_in;
```

Comment:
new?

Reply:
This is used to check if the previous state from the current state. Have a look at the line 1215 if ... notice the if statement. When the state has changed, then we will update the database. However, if they state has **not** changed, then we don't need to update the database.